src/Microsoft.ML.Data/Commands/CrossValidationCommand.cs

```diff
@@ -332,7 +332,7 @@ private string GetSplitColumn(IChannel ch, IDataView input, ref IDataView output
                     var hashargs = new HashTransform.Arguments();
                     hashargs.Column = new[] { new HashTransform.Column { Source = origStratCol, Name = stratificationColumn } };
                     hashargs.HashBits = 30;
-                    output = new HashTransform(Host, hashargs, input);
+                    output = HashTransform.Create(Host, hashargs, input);
```

Comment:
>Create [](start = 43, length = 6)

Can we try to limit the use of `Create` to dependency injection? You should just do `new HashTransform(Host, origStratCol, stratificationColumn, hashBits: 30).Transform(input)`

Reply:
instead of lines 332-335

---
In reply to: [218911189](https://github.com/dotnet/machinelearning/pull/944#discussion_r218911189) [](ancestors = 218911189)